### PR TITLE
CORTX-32157: Refactor Server chart values

### PR DESCRIPTION
**Pull request by keithpine**
## Description

Cleanup Server chart values, in the same vein as previous changes.

- Rename object from cortxserver to server
- Use camel case for variable names
- etc.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds new functionality)
- [ ] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [ ] Documentation additions or improvements
- [ ] Code quality improvements to existing code or test additions/updates

## Applicable issues

- This change fixes an issue: CORTX-32157
- This change is related to an issue: #

## How was this tested?

Deployed cluster, ran scripts, S3 I/O, CSM APIs

Compared the output of `kubkect get deployments,statefulsets,services,cm -o yaml` from before and after this change, it is identical, minus the unique deployment metadata and the ordering change betwen cortx-rgw and cortx-hax containers.

## Additional information

A significant part of this change is defining new template functions for declaring the cortx-hax and cortx-setup cluster. I got a bit tired of copying and pasting the new image definitions. Since these containers are identical for all Pods, it's simple to declare and define as template functions. This reduces a lot of the code duplication between workload definitions.

This works well as long as all the hax/setup container definitions are identical.

A downside is that it obscures the contents of the Pod specs, so you don't see at a glance what is being defined. A call to `helm template` easily displays this. However, if you don't feel this is a good improvement, feel free to commend and I can revert back to the full definitions.

## Checklist

- [x] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [x] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change requires newer CORTX or third party image versions:

- [ ] The `image` fields in [solution.example.yaml](../k8_cortx_cloud/solution.example.yaml) have been updated to use the required versions.
- [ ] The `appVersion` field of the [Helm chart](../charts/cortx/Chart.yaml) has been updated to use the new CORTX version.

If this change addresses a CORTX Jira issue:

- [x] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)

-----
[View rendered charts/cortx/README.md](https://github.com/keithpine/cortx-k8s/blob/CORTX-32157_server-refactor/charts/cortx/README.md)

### CODE REVIEW

File: charts/cortx/values.yaml [hunk 1]
```diff
@@ -232,49 +233,88 @@ cortxha:
...
+    size: 1Gi
+    # -- Persistent volume access modes
+    accessModes:
+      - ReadWriteOnce
+
+  auth:
+    # -- Name of the admin user to create when initializing the cluster
+    adminUser: cortx-admin
+    # -- Access Key of the admin user
+    adminAccessKey: cortx-user
```
Comment (keithpine) [inline charts/cortx/values.yaml:303]: Looks like I have these backwards. They are written correctly to the config.yaml, but the descriptions are reversed.

Comment (keithpine) [reply charts/cortx/values.yaml:303]: Also, how do we feel about the naming of these?

`authUser` (RGW admin user name) and `authAdmin` (RGW admin user Access Key) don't make much sense to me.

File: charts/cortx/values.yaml [hunk 2]
```diff
@@ -232,49 +233,88 @@ cortxha:
...
+
+  # TODO: can this be removed?
+  maxStartTimeout: 240
```
[the remark this answers is not in the document]
Comment (walterlopatka) [reply charts/cortx/values.yaml:307]: Sounds good 👍

**Review by walterlopatka — APPROVED** [review]
lgtm

File: charts/cortx/templates/_images.tpl [hunk 3]
```diff
@@ -13,9 +13,103 @@ Return the Control image name
...
 */}}
 {{- define "cortx.client.image" -}}
 {{ include "cortx.images.image" (dict "image" .Values.client.image "root" .) }}
 {{- end -}}
+
+{{/*
+Return the CORTX setup initContainer
+{{ include "cortx.containers.setup" ( dict "image" .Values.path.to.the.image "root" $) }}
+*/}}
+{{- define "cortx.containers.setup" -}}
```
[the remark this answers is not in the document]
Comment (osowski) [reply charts/cortx/templates/_images.tpl:34]: That's what I was thinking but also throwing that in it's standalone template that recursively calls this template. I'm good either way, just wanted to try to make it so the Data Pods weren't still creating their own initContainer from scratch when all the other Pods were using the template form.

**Review by osowski — APPROVED** [review]
lgtm

**Pull request merged**